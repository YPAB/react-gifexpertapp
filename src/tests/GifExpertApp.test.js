import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import GitExpertApp from '../GifExpertApp';

describe('Pruebas en el componente <GitExpertApp/>',() => {

    

    test('Debe crear una copia de GitExpertApp', () => {

        const wrapper = shallow (<GitExpertApp/>);

        expect(wrapper).toMatchSnapshot();


    });


    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['Homero','Los Simpson'];
        const wrapper = shallow (<GitExpertApp defaultCategories={categories }/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);


    });

})