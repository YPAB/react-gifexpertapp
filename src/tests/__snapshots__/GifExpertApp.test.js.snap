// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GitExpertApp/> Debe crear una copia de GitExpertApp 1`] = `
<Fragment>
  <h2
    className="animate__wobble"
  >
    Aplicación Gif - Udemy
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el componente <GitExpertApp/> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2
    className="animate__wobble"
  >
    Aplicación Gif - Udemy
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Homero"
      key="Homero"
    />
    <GifGrid
      category="Los Simpson"
      key="Los Simpson"
    />
  </ol>
</Fragment>
`;
